### PR TITLE
[BEAM-475] Enable javadoc for all modules by default

### PR DESCRIPTION
**Pull request by dhalperi**
And fix several issues in several modules that were improperly tested before.

R: @davorbonaci

### CODE REVIEW

File: examples/java8/src/main/java/org/apache/beam/examples/complete/game/LeaderBoard.java [hunk 1]
```diff
@@ -73,7 +73,7 @@
  * results, e.g. for 'team prizes'. We're now outputting window results as they're
  * calculated, giving us much lower latency than with the previous batch examples.
  *
- * <p>Run {@link injector.Injector} to generate pubsub data for this pipeline.  The Injector
```
Comment (dhalperi) [inline examples/java8/src/main/java/org/apache/beam/examples/complete/game/LeaderBoard.java]: file JIRA: the injector is still referred to in some examples.

File: runners/core-java/pom.xml [hunk 2]
```diff
@@ -65,29 +65,6 @@
 
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-javadoc-plugin</artifactId>
```
Comment (dhalperi) [inline runners/core-java/pom.xml]: replaced by my open PR for javadoc aggregation.

File: runners/google-cloud-dataflow-java/pom.xml [hunk 3]
```diff
@@ -85,76 +85,6 @@
...
-
-          <subpackages>org.apache.beam.runners.dataflow</subpackages>
-          <additionalparam>-exclude org.apache.beam.sdk.runners.dataflow.internal:org.apache.beam.sdk.runners.dataflow.testing:org.apache.beam.sdk.runners.dataflow.util ${beam.javadoc_opts}</additionalparam>
-          <use>false</use>
-          <quiet>true</quiet>
-          <bottom><![CDATA[<br>]]></bottom>
-
-          <offlineLinks>
-            <offlineLink>
-              <url>https://developers.google.com/api-client-library/java/google-api-java-client/reference/1.20.0/</url>
```
Comment (dhalperi) [inline runners/google-cloud-dataflow-java/pom.xml]: TODO: figure out where these connections go.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/IsmFormat.java [hunk 4]
```diff
@@ -167,11 +167,12 @@ public V getValue() {
     }
   }
 
-  /** A {@link Coder} for {@link IsmRecord}s.
+  /**
+   * A {@link Coder} for {@link IsmRecord}s.
    *
    * <p>Note that this coder standalone will not produce an Ism file. This coder can be used
    * to materialize a {@link PCollection} of {@link IsmRecord}s. Only when this coder
-   * is combined with an {@link IsmSink} will one produce an Ism file.
```
Comment (dhalperi) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/IsmFormat.java]: IsmSink is not in the Beam SDK

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java [hunk 5]
```diff
@@ -76,15 +76,15 @@
  *
  * <p>Example usage:
  *
- * <pre> {@code
+ * <pre>{@code
  * PCollection<String> lines = ... ;
  * PCollection<String> words =
  *     lines.apply(ParDo.of(new DoFn<String, String>() {
- *         @ProcessElement
+ *         {@verbatim @}ProcessElement
```
Comment (davorbonaci) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java]: literal? (here and elsewhere)

**Pull request closed without merging**